middlewares/accesslog/logger_test.go

```diff
@@ -68,10 +69,55 @@ func TestLogger(t *testing.T) {
 
 	logger.ServeHTTP(&logtestResponseWriter{}, req, LogWriterTestHandlerFunc)
 
-	if logdata, err := ioutil.ReadFile(logfilePath); err != nil {
+	if logdata, err := ioutil.ReadFile(logfilePath); err == nil {
+		assertValidLogData(t, logdata)
+	} else {
 		fmt.Printf("%s\n%s\n", string(logdata), err.Error())
 		assert.NoError(t, err)
-	} else if tokens, err := shellwords.Parse(string(logdata)); err != nil {
+	}
+}
+
+func TestNewLogHandlerOutputStdout(t *testing.T) {
+	file, err := ioutil.TempFile("", "testlogger")
+	if err != nil {
+		t.Fatalf("failed to create temp file: %s", err)
+	}
+	defer file.Close()
+	// capture stdout in a file
+	stdOut := os.Stdout
+	os.Stdout = file
```

Comment:
I also checked and saw no "native" way of doing this, whereas the current approach with `TempFile` is included. Therefore I would stick with the current solution.